src/app/_components/RolesList/RolesList.tsx

```diff
@@ -1,43 +1,45 @@
 'use client';
 
-import {useMemo} from "react";
-import {Roles} from "@/roles";
-import {Modifiers} from "@/modifiers";
-import {RoleOrModifierTypes} from "@/constants/rolesAndModifiers";
+import {type FC, useMemo} from "react";
+import {type Modifier, type Role} from "@/constants/rolesAndModifiers";
 import {Teams} from "@/constants/teams";
 import {RolesListContext, TeamFilters, TypeFilters, useFilters, useSearch} from "./hooks";
 import {Search} from "./Search";
 import {Filters} from "./Filters";
 import {RoleCard} from "./RoleCard/RoleCard";
 
-const RolesAndModifiers = [...Roles, ...Modifiers];
-
-export const RolesList = ({ hideElement = false }: { hideElement?: boolean }) => {
+export const RolesList: FC<{
+    roles: Role[],
+    modifiers: Modifier[],
+    hideSettings?: boolean
+    hideTips?: boolean
+}> = ({roles, modifiers, hideSettings = false, hideTips = false}) => {
     const { searchValue, search } = useSearch();
     const { typeFilterValue, teamFilterValue, filter } = useFilters();
 
     const results = useMemo(
         () => {
-            let rolesAndModifiers = RolesAndModifiers;
-
-            if (searchValue) {
-                rolesAndModifiers = rolesAndModifiers.filter(
-                    ({name}) =>
-                        name.toLowerCase().includes(searchValue.toLowerCase())
-                )
-            }
+            let rolesAndModifiers;
 
             switch (typeFilterValue) {
                 case TypeFilters.Role:
-                    rolesAndModifiers = rolesAndModifiers.filter(({type}) => type === RoleOrModifierTypes.Role);
```

Comment:
To jest przykład fajnej optymalizacji - wcześniej ładowaliśmy tablicę ról razem z modyfikatorami i na niej przeprowadzaliśmy operację filter. Optymalizacją tutaj jest pozbycie się filtra i działanie na danym zbiorze od razu zależnie od danego typu.